server/src/main/java/org/cloudfoundry/identity/uaa/scim/vmidentity/VmidentityScimUserProvisioning.java

```diff
@@ -104,115 +111,59 @@ public ScimUser delete(String id, int version) {
     }
 
     @Override
-    public List<ScimUser> query(String filter)
```

Comment:
let's add a todo: for considering improving perf for cases when filter is used for user lookup (i.e. filter is user_id=upn) and submit....